packages/ragbits-document-search/src/ragbits/document_search/documents/document.py

```diff
@@ -50,6 +60,19 @@ def id(self) -> str:
         """
         return self.source.id
 
+    def add_location_metadata(self, provider_metadata: dict | None) -> None:
```

Comment:
why `| None` here?

Reply:
in unstructured output format metadata was marked as optional AFAIR

Reply:
I don't see optional typing for `element.metadata` anywhere, but still you use this method with `model_dump` in the condition, so you're sure it's a dict here